iroh-bytes/src/store/file.rs

```diff
@@ -1602,10 +1602,16 @@ impl ActorState {
                             })?
                             .to_owned();
                         // we can not reference external files, so we just copy them. But this does not have to happen in the actor.
-                        self.rt.spawn_blocking(move || {
-                            tx.send(export_file_copy(temp_tag, path, size, target, progress))
-                                .ok();
-                        });
+                        if path == target {
```

Comment:
Then you would break it as well. But that's on you then. When using external files the user promises to keep the files unchanged.

If you do
`export <hash1> foo.bin`
`export <hash2> foo.bin`

you are basically violating that promise.

Reply:
I know it is a very poor man solution, but may it worth to add nested size comparision here to detect 99.99% of such cases and add warning?

Reply:
no, there are a million ways how you can modify files and void the contract. This would just avoid one, and only in some cases.

what you could do is to check if the target is in the set of all external files and in that case refuse. It would be costly though, and it would still not help with symlinks and hardliks.

Reply:
I meant cases when you export file to existing file path, not all possible cases. I can easily imagine cases when people do something like `iroh export-file <doc>/<very-complicated-key-name>.zip ~/archive.zip` with different files and warning would cover most of such cases.

Reply:
I meant cases when you export file to existing file path, not all possible cases. I can easily imagine cases when people do something like iroh export-file <doc>/<very-complicated-key-name>.zip ~/archive.zip with different files and warning would cover most of such cases.

Reply:
How about just a warning if you are going to override something? Would make sense in any case, and when you use it from code you would not pay for it.

Reply:
Yep, warning would be nice